### PR TITLE
Improved Documentation, including providing the first version of Spark-Redis that createRedisXStream was made available (2.3.1) and fixing grammar/spelling.

### PR DESCRIPTION
**Pull request by justinrmiller**

### CODE REVIEW

File: doc/cluster.md [hunk 1]
```diff
@@ -5,15 +5,15 @@ def twoEndpointExample ( sc: SparkContext) = {
...
   val rddFromEndpoint2 = {
-    //endpoint("127.0.0.1", 7379) as the default connection in this block
+    // endpoint("127.0.0.1", 7379) as the default connection in this block
     implicit val c = redisConfig2
     sc.fromRedisKV("*")
   }
 }
 ```
-If you want to use multiple redis clusters/instances, an implicit RedisConfig can be used in a code block to specify the target cluster/instance in that block.
+If you want to use multiple Redis clusters/instances, an implicit RedisConfig can be used in a code block to specify the target cluster/instance in that block.
```
Comment (justinrmiller) [inline doc/cluster.md]: "in that block" may be redundant here, but I wasn't sure if the author had another meaning for "block" in the second usage.

Comment (fe2s) [reply doc/cluster.md]: I think it's redundant and can be removed.

File: doc/dataframe.md [hunk 2]
```diff
@@ -36,7 +36,7 @@ object DataFrameExample {
       .config("spark.redis.port", "6379")
       .getOrCreate()
 
-    val personSeq = Seq(Person("John", 30), Person("Peter", 45))
+    val personSeq = Seq(Person("John", 30), Person("Peter", 45)
```
Comment (fe2s) [inline doc/dataframe.md]: Looks like a typo, shouldn't be any changes on this line

File: doc/dataframe.md [hunk 3]
```diff
@@ -154,17 +154,16 @@ df.write
...
 
-
 ### Persistence model
 
-By default DataFrames are persisted as Redis Hashes. It allows to write data with Spark and query from non-Spark environment.
+By default DataFrames are persisted as Redis Hashes. It allows for data to be written with Spark and queried from a non-Spark environment.
 It also enables projection query optimization when only a small subset of columns are selected. On the other hand, there is currently 
-a limitation with Hash model - it doesn't support nested DataFrame schema. One option to overcome it is making your DataFrame schema flat.
-If it is not possible due to some constraints, you may consider using Binary persistence model.
+a limitation with the Hash model - it doesn't support nested DataFrame schemas. One option to overcome this is to your DataFrame schema flat.
```
Comment (fe2s) [inline doc/dataframe.md]: `"One option to overcome this is to your DataFrame schema flat." `
should it be
"One option to overcome this is to make your DataFrame schema flat." ?

File: doc/cluster.md (hunk 1 above)
Comment (justinrmiller) [reply doc/cluster.md]: Sounds good!

File: doc/dataframe.md (hunk 2 above)
Comment (justinrmiller) [reply doc/dataframe.md]: Ah I thought I was eliminating a newline of that parenthesis, will fix.

File: doc/dataframe.md (hunk 3 above)
Comment (justinrmiller) [reply doc/dataframe.md]: Yep, fixed.

**Pull request merged**